schemas/constructs/v1alpha1/capability.json

```diff
@@ -0,0 +1,88 @@
+{
+  "$id": "https://schemas.meshery.io/capability.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "Meshery manages entities in accordance with their specific capabilities. This field explicitly identifies those capabilities largely by what actions a given component supports; e.g. metric-scrape, sub-interface, and so on. This field is extensible. Entities may define a broad array of capabilities, which are in-turn dynamically interpretted by Meshery for full lifecycle management.",
+  "additionalProperties": false,
+  "type": "object",
+
+  "required": ["schemaVersion", "version", "displayName", "kind", "type", "state", "status"],
+
+  "properties": {
+    "schemaVersion": {
+      "$ref": "../core.json#/definitions/versionString",
+      "description": "Specifies the version of the schema to which the capability definition conforms."
+    },
+    "version": {
+      "$ref": "../core.json#/definitions/semverString",
+      "description": "Version of the capability definition.",
+      "minLength": 1,
+      "maxLength": 50
+    },
+    "displayName": {
+      "$ref": "../core.json#/definitions/inputString",
+      "description": "Name of the capability in human-readible format."
+    },
+    "description": {
+      "type": "string",
+      "description": "A written representation of the purpose and characteristics of the capability."
+    },
+
+    "kind": {
+      "$ref": "../core.json#/definitions/inputString",
+      "description": "Top-level categorization of the capability",
+      "anyOf": [
+        {
+          "const": "action",
+          "description": "For capabilities related to executing actions on entities. Example: initiate log streaming on a Pod. Example: initiate deployment of a component."
+        },
+        {
+          "const": "mutate",
+          "description": "For capabilities related to mutating an entity. Example: the ability to change the configuration of a component."
+        },
+
+        {
+          "const": "interaction",
+          "description": "Catch all for capabilities related to interaction with entities. Example: the ability for a component to be dragged and dropped. Example: supports event bubbling to parent components. "
+        }
+      ]
+    },
+
+    "type": {
+      "$ref": "../core.json#/definitions/inputString",
+      "description": "Classification of capabilities. Used to group capabilities similar in nature."
+    },
+
+    "subType": {
+      "$ref": "../core.json#/definitions/inputString",
+      "description": "Most granular unit of capability classification. The combination of Kind, Type and SubType together uniquely identify a Capability."
+    },
+
+    "key": {
+      "$ref": "../core.json#/definitions/inputString",
```

Comment:
As combination of kind, type and subType together uniquely identify a capability, so do we need this?

We can have key identified / discerned as `kind_type_subType`.

Reply:
@MUzairS15 this is more or less a placeholder for future incorporation of permissions.